### PR TITLE
Documentation for the callbacks

### PR DESCRIPTION
**Pull request by cmrd-senya**

### CODE REVIEW

File: lib/diaspora_federation.rb [hunk 1]
```diff
@@ -14,6 +14,63 @@
...
 # diaspora* federation library
+#
+# In order to communicate with the application which uses the diaspora_federation gem
+# callbacks are introduced. The callbacks are used for getting required data from the
+# application or posting data to the application.
+#
+# Callbacks are implemented at the application side and must follow these specifications:
+#
+# fetch_person_for_webfinger
+#   Fetches person data from the application in a WebFinger document format
```
Comment (SuperTux88) [inline lib/diaspora_federation.rb]: it's data for the WebFinger, not "WebFinger document format" (which is xml/json), same is for  HCard, it's for the HCard, but not jet HCard format.

**Comment by SuperTux88** [comment]
I think I would move the whole stuff to the `define_callbacks` method.

File: lib/diaspora_federation.rb [hunk 2]
```diff
@@ -14,6 +14,63 @@
...
+#   @return [DiasporaFederation::Discovery::WebFinger] webfinger data
+#
+# fetch_person_for_hcard
+#   Fetches person data from the application in a HCard document format
+#   @param [String] guid of the person
+#   @return [DiasporaFederation::Discovery::HCard] webfinger data
+#
+# save_person_after_webfinger
+#   After the gem had made a person discovery using WebFinger it calls this callback
+#   so the application saved the person data
```
Comment (SuperTux88) [inline lib/diaspora_federation.rb]: "saves" not "saved"

**Comment by cmrd-senya** [comment]
Here you are, sir :)

**Comment by SuperTux88** [comment]
thanks! :+1:

**Comment by cmrd-senya** [comment]
you are welcome :)

**Pull request merged**